yateto/ast/node.py

```diff
@@ -446,3 +446,37 @@ def __str__(self):
     Bstr = self.indexString('B', [self._k, self._n], self.rightTerm().indices, self._transB)
     Cstr = self.indexString('C', [self._m, self._n], self.indices)
     return '{} [{}]: {} = {} {}'.format(type(self).__name__, self.indices, Cstr, Astr, Bstr)
+
+  def is_gemm(self):
+    left_indices = self.leftTerm().indices
+    right_indices = self.rightTerm().indices
+    if not (len(left_indices) == 2 and len(right_indices) == 2):
```

Comment:
yep. I can see your point.  it will work if free indices are nicely grouped and do not require an exchange (which, as far as I remember, always required for efficient tensor contractions to avoid transpositions as much as possible). Could I rename `is_gemm` to `is_pure_gemm` for now? A tensor support is something that I will address in a separate PR.